cloudify/plugins/workflows.py

```diff
@@ -76,6 +78,132 @@ def auto_heal_reinstall_node_subgraph(
                                        ignore_failure=ignore_failure)
 
 
+def get_groups_with_members(ctx):
+    groups_members = {}
+    for instance in ctx.node_instances:
+        scaling_group_ids = [sg.get('id') for sg in instance.scaling_groups]
+        for sg_id in scaling_group_ids:
+            if sg_id not in groups_members:
+                groups_members[sg_id] = []
+            groups_members[sg_id].append(instance.id)
+    return groups_members
+
+
+def _check_for_too_many_exclusions(exclude_instances, available_instances,
```

Comment:
please docstring what does this check, and what are the args (string - ids? or instances, or dicts)